scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -823,13 +824,30 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       if (isBracket) close // If we can fit the type params, make it so
       else lastParen // If we can fit all in one block, make it so
 
-    Seq(
+    // Determine if we should just allocate a line for each parameter
+    // definition for case classes, and not attempt to fit everything
+    // on a single line.
+    val newlineBefore =
+      !isBracket && style.newlines.beforeCaseClassParameterDefn && owners(
+        ft.left).is[CaseClass]
+
+    val fitInSingleLine =
       Split(NoSplit, 0)
-        .withPolicy(SingleLineBlock(singleLineExpire)),
```

Comment:
I think you can use `ignoreIf = ...` to conditionally skip `fitInSingleLine` if this setting is enabled.

Reply:
That should make the diff for this PR very small.